app/models/user.rb

```diff
@@ -7,4 +9,35 @@ class User < ApplicationRecord
   validates :password, presence: true, length: { minimum: 6 }
 
   has_secure_password
+
+  def remember
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+    remember_digest
+  end
+
+  def session_token 
+    remember_digest || remember 
+  end
+
+  def authenticated?(remember_token)
+    return false if remember_digest.nil?
+
+    BCrypt::Password.new(remember_digest).is_password?(remember_token)
+  end
+
+  def forget 
```

Comment:
I know this was not communicated before, you're learning it now :P: when a method changes the internal state of the parent object, you should put a `!` to it. See more: http://ruby-for-beginners.rubymonstas.org/objects/bangs.html

Reply:
Done!